middleware/validator.js

```diff
@@ -0,0 +1,54 @@
+"use strict";
+
+const schema = {
+    name: 'string',
+    details: {
+        url: 'string',
+        description: 'string'
+    }
+};
+
+function validate(obj, schema, res){
+    return Object.keys(obj).every(function(element) {
+        if (schema.hasOwnProperty(element)) {
```

Comment:
Try to avoid using `else` where it is possible.
Create variable before `if` with default value
Change it in `if`, 
And return after all